.github/workflows/queasars_deployment.yml

```diff
@@ -82,13 +85,17 @@ jobs:
     # Only deploy to PyPI if both the project version is consistent and the CI is run successfully.
     name: Deploy to PyPI
     needs: [Check_Version_Consistency, Run_CI]
-    if: ${{ inputs.deploy_to_pypi == '' || inputs.deploy_to_pypi == true }}
+    if: ${{ github.event_name == 'push' || inputs.deploy_to_pypi }}
     uses: ./.github/workflows/queasars_deploy_to_pypi.yml
     secrets: inherit
 
   Deploy_Documentation:
     # Only deploy to GitHub Pages if the deployment to PyPI succeeded.
     name: Deploy to Github Pages
     needs: Deploy_To_PyPI
-    if: ${{ inputs.deploy_to_pages == '' || inputs.deploy_to_pages == true }}
+    if: |
+      always() &&
+      !cancelled() &&
+      !failure() &&
+      ( github.event_name == 'push' || inputs.deploy_to_pages )
```

Comment:
Why the three conditions on top? Does the job get executed otherwise even when the workflow is "canceled"?

Reply:
Ok, this one is a little complex to explain. Generally, we want the documentation deployment to only deploy, once the deployment to PyPI has succeeded. This is expressed by the needs job relationship. The problem is, that this relationship is disturbed, when the PyPI deployment is skipped by it's if condition. Consequently in this case, the documentation deployment doesn't run, even if we want it to. To counter that, we have to use always() to force the documentation to deploy. To then prevent it from deploying, when we don't want it to, we then have to specify, that it should not deploy, when previous jobs were failed or cancelled or if it is not meant to run in manual deployment. More discussions about this can be found in actions/runner#491 .

Reply:
Thanks for explaining!